python/ray/serve/controller.py

```diff
@@ -424,7 +424,8 @@ def deploy(
     ) -> bool:
         """Deploys a deployment."""
         if route_prefix is not None:
-            assert route_prefix.startswith("/")
+            if not route_prefix.startswith("/"):
+                raise ValueError("route_prefix must starts with a /")
         if docs_path is not None:
             assert docs_path.startswith("/")
```

Comment:
@edoakes let me know if I should patch this as well. I can open another PR for it

Reply:
I think we should :)

Reply:
I think this is not the right place to raise the user-facing error -- we can do it on the client in `serve.run` to avoid having a nasty `RayTaskError` traceback

Reply:
Oh ok, that makes sense. But in that case would it be better if we just append a `/` prefix or we still want to throw the ValueError?